app/gateway/2.8.x/admin-api/developers/reference.md

```diff
@@ -0,0 +1,1439 @@
+    "key": "wpJqvsTFRjgQstL16pQTYKtlHfkzdJpb"
+}
+```
+
+### Update a credential
+
+**Endpoint**
+
+<div class="endpoint patch">/developers/{DEVELOPER_EMAIL|DEVELOPER_ID}/credentials/{PLUGIN_NAME}/{CREDENTIAL_ID}</div>
+
+Attribute                         | Description
+---------:                        | --------   
+`{DEVELOPER_EMAIL|DEVELOPER_ID}`  | The email or UUID of a developer.
+`{PLUGIN_NAME}` | The name of a supported authentication plugin. Can be one of: <br> &#8226; `basic-auth` <br> &#8226; `oauth2` <br> &#8226; `hmac-auth` <br> &#8226; `jwt` <br> &#8226; `key-auth` <br> &#8226; `openid-connect`
+`{CREDENTIAL_ID}` | The UUID of a credential for the plugin.
+
+**Request body**
+
+Attribute                    | Description
+---------:                   | --------   
+Plugin configuration fields | Any authentication credentials for the plugin that you are configuring. See the plugin's reference on the [Plugin Hub](/hub/) to find the parameters for a specific plugin.
+
+Example request to update a `key-auth` key:
+```
+http PATCH :8001/developers/91a1fb59-d90f-4f07-b609-4c2a1e3d847e/credentials/key-auth/1b0ba8f4-dd9f-4f68-8108-c81952e05b8e \
+  key=apikey
+```
+
+**Response**
+```
+HTTP/1.1 200 OK
+```
+
+The exact response depends on the plugin. For example, if you update a `key-auth`
+credential to use a custom key instead of a UUID:
+
+```json
+{
+    "consumer": {
+        "id": "bc504999-b3e5-4b08-8544-86962c969335"
+    },
+    "created_at": 1645567499,
+    "id": "1b0ba8f4-dd9f-4f68-8108-c81952e05b8e",
+    "key": "apikey"
+}
+```
+
+### Delete a credential
+
+**Endpoint**
+
+<div class="endpoint delete">/developers/{DEVELOPER_EMAIL|DEVELOPER_ID}/credentials/{PLUGIN_NAME}/{CREDENTIAL_ID}</div>
+
+Attribute                         | Description
+---------:                        | --------   
+`{DEVELOPER_EMAIL|DEVELOPER_ID}`  | The email or UUID of a developer.
+`{PLUGIN_NAME}` | The name of a supported authentication plugin. Can be one of: <br> &#8226; `basic-auth` <br> &#8226; `oauth2` <br> &#8226; `hmac-auth` <br> &#8226; `jwt` <br> &#8226; `key-auth` <br> &#8226; `openid-connect`
+`{CREDENTIAL_ID}` | The UUID of a credential for the plugin.
+
+**Response**
+```
+HTTP/1.1 204 No Content
+```
+
+## Application authentication
+
+### Inspect all credentials for an application
+
+Get a list of all application credentials for a specific type of authentication
+plugin.
+
+**Endpoint**
+
+<div class="endpoint get">/developers/{DEVELOPER_EMAIL|DEVELOPER_ID}/applications/{APPLICATION_ID}/credentials/{PLUGIN_NAME}</div>
+
+Attribute                         | Description
+---------:                        | --------   
+`{DEVELOPER_EMAIL|DEVELOPER_ID}`  | The email or UUID of a developer.
+`{APPLICATION_ID}`  | The application UUID.
+`{PLUGIN_NAME}` | The authentication strategy in use for application registration. Can be one of: <br>&#8226; `oauth2` <br> &#8226; `key-auth` <br> &#8226; `openid-connect`
```

Comment:
Unsure if correct. Went with authentication strategies available for app registration, as documented here: https://docs.konghq.com/gateway/2.7.x/developer-portal/administration/application-registration/auth-provider-strategy/

Reply:
My understanding of this after trying some things out is that `ouath2` is the only valid value here anymore. It is related to the `portal_app_auth` kong.conf setting: https://github.com/Kong/kong-ee/blob/master/kong.conf.default#L2062

If `portal_app_auth = kong-oauth2` then you can `oauth2` as the `{PLUGIN_NAME}`.
If `portal_app_auth = external-oauth2` then I don't believe you can use this endpoint.

This is just based on me learning this now and testing out the combinations. I'm unsure if anyone else has knowledge about this and can chime in.

Reply:
That was my experience as well, though I couldn't figure out what happens for the other options. It's probably safe to say, for now, that this endpoint requires `portal_app_auth = kong-oauth2`.